src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -1046,6 +1046,11 @@ impl<E: Encoder> InvertedIndex<E> {
             )
         }
     }
+
+    /// Create a new [`IndexReader`] for this inverted index.
+    pub fn reader<D: Decoder>(&self, decoder: D) -> IndexReader<'_, D> {
```

Comment:
the `decoder` is a bit redudant here as we already have the encoder.

@chesedo : IIRC we discussed at some point to have API mapping encoder <-> decoder. Maybe we should add that?

Reply:
Yes, this will work. We can figure out how to give it filter arguments later